### PR TITLE
Port to mozjs78

### PR DESCRIPTION
**Pull request by mtwebster**

### CODE REVIEW

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- js/perf/core.js  3
         

Complexity decreasing per file
==============================
+ js/ui/overrides.js  -23
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6310037&bid=20517494)

**Comment by Fantu** [comment]
@mtwebster thanks, from a fast look dump of libcjs-dev build-dep to a version that have rebased cjs (for example >=4.7 or >=4.8 based on cjs version update) is missed  (to avoid try to build with older version)
if you want take a look in debian there was patch (including leigh123linux patch) already tested and working (to check if there are something different and useful): https://salsa.debian.org/cinnamon-team/cinnamon/-/commit/dfd15d78b818cc9d36995a07a86dc68bc0d24123
and the other with use of gjs instead (also this tested with debian unstable and working): https://salsa.debian.org/cinnamon-team/cinnamon/-/commit/9f21e2426c3325ab8b67de40a474166618d6cec0

**Comment by mtwebster** [comment]
I think I got pretty much everything in that cinnamon patch (those originated from the individual patches I have here). I'll go thru them again to double check.

What do you mean with the libcjs-dev, You mean I should add a minimum version for 'cjs' here? https://github.com/linuxmint/cjs/pull/84/commits/8ef6d459967348b396d2dac500430c8cc8797283#diff-3a8967c9c7449446ddcdc76faa20bfafR71

**Comment by Fantu** [comment]
from your link with cjs rebased on version 4.6.1, on this PR you should add change of "libcjs-dev (>= 3.8)," to libcjs-dev (>= 4.6.1)," in debian/control and for other distros is good also bump it in configure.ac changing "GJS_MIN_VERSION=3.2.0" to "GJS_MIN_VERSION=4.6.1"
sry for my bad english and if I haven't been able to explain myself

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- js/perf/core.js  3
         

Complexity decreasing per file
==============================
+ js/ui/overrides.js  -23
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6310037&bid=20517494)

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- js/perf/core.js  3
         

Complexity decreasing per file
==============================
+ js/ui/overrides.js  -23
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6310037&bid=20517494)

File: js/ui/backgroundManager.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const Gio = imports.gi.Gio;
 const Lang = imports.lang;
 const Mainloop = imports.mainloop;
 
-class BackgroundManager {
+var BackgroundManager = class {
```
Comment (kungfooman) [inline js/ui/backgroundManager.js:7]: Can you give the reason why this is important? Dynamic reloading / overwrite at runtime?

**Pull request merged**